tests/metrics-server_resource_table.py

```diff
@@ -0,0 +1,338 @@
+                for container in containers:
+                    requests = container.get('resources', {}).get('requests', {})
+                    cpu_request = requests.get('cpu', '0')
+                    memory_request = requests.get('memory', '0')
+                    
+                    if cpu_request != '0' or memory_request != '0':
+                        cpu_millicores = parse_resource_value(cpu_request, 'cpu')
+                        memory_mib = parse_resource_value(memory_request, 'memory')
+                        
+                        component_requests[component]['cpu'] += cpu_millicores
+                        component_requests[component]['memory'] += memory_mib
+        
+        except Exception:
+            continue  
+    
+    return component_requests, dict(component_storage)
+
+def calculate_max_resources(actual_usage, manifest_requests):
+    """Calculate maximum of actual usage and manifest requests"""
+    all_components = set(actual_usage.keys()) | set(manifest_requests.keys())
+    max_resources = defaultdict(lambda: {'cpu': 0, 'memory': 0})
+    
+    for component in all_components:
+        actual = actual_usage.get(component, {'cpu': 0, 'memory': 0})
+        requests = manifest_requests.get(component, {'cpu': 0, 'memory': 0})
+        
+        max_resources[component]['cpu'] = max(actual['cpu'], requests['cpu'])
+        max_resources[component]['memory'] = max(actual['memory'], requests['memory'])
+    
+    return max_resources
+
+def generate_table(component_resources, storage_map, actual_usage, manifest_requests):
+    """Generate markdown table from component resources"""
+    print("## Resource Usage by Components")
+    print()
+    print("The following table shows the resource requirements for each Kubeflow component:")
+    print()
+    print("| Component | CPU (cores) | Memory (Mi) | Storage (GB) |")
+    print("|-----------|-------------|-------------|--------------|")
+    
+    totals = {'cpu': 0, 'memory': 0, 'storage': 0}
+    
+    for component in sorted(component_resources.keys()):
+        resources = component_resources[component]
+        storage = storage_map.get(component, 0)
+        
+        totals['cpu'] += resources['cpu']
+        totals['memory'] += resources['memory']
+        totals['storage'] += storage
+        
+        print(f"| {component} | {resources['cpu']}m | {resources['memory']}Mi | {storage}GB |")
+    
+    print(f"| **Total** | **{totals['cpu']}m** | **{totals['memory']}Mi** | **{totals['storage']}GB** |")
+    print()
+    
+    print("- **Other**: Miscellaneous Kubeflow components including:")
+    print("  - PVC Viewer Controller")
+    print("  - Tensorboard Controller and Web App")
+    print("  - Multi-tenancy and RBAC components")
+    print("  - Network policies")
+    print("  - Custom resources and CRDs")
+    print("  - User namespace resources")
+    print("  - Additional Kubeflow utilities and tools")
```

Comment:
We have the components here https://github.com/kubeflow/manifests#kubeflow-components-versions . It should be covered in the same detail and not under other. i would even just extend this table with more columns and remove https://github.com/kubeflow/manifests#resource-usage-by-components

Reply:
Please update the table in the main readme after fixing the other open issues